compression/lempel_ziv.py

```diff
@@ -22,3 +25,26 @@ def inverse_lz77(data):
         text, length = text + text[-offset:], length - offset
       text += text[-offset:-offset + length] + c
   return text
+
+def naive_factorization(w, n):
+  LZ = []
+  i = 1
+  while i <= n:
+    matched = [w[i:(i + k)]
```

Comment:
You can write w[i:i + k], : has lower precendence than + if I remember

Reply:
Ack